### PR TITLE
[Improvement]: Change default behavior of fallback documents

### LINKED ISSUE #10346: [Improvement]: Change default behavior of fallback documents
### Improvement description

related to #10005 and #10163

It's actually about removing the config option `allow_processing_unpublished_fallback_document` and changing the default behavior.

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves #10346

### CODE REVIEW

File: bundles/CoreBundle/src/EventListener/Frontend/ElementListener.php [hunk 1]
```diff
@@ -97,22 +95,6 @@ public function onKernelController(ControllerEvent $event)
                     $document->getFullPath(),
```
Comment (mattamon) [inline bundles/CoreBundle/src/EventListener/Frontend/ElementListener.php]: What about this line above? Can't the !$request->attributes->get(self::FORCE_ALLOW_PROCESSING_UNPUBLISHED_ELEMENTS) be removed as well?
```php
   if ($document && !$document->isPublished() && !$user && !$request->attributes->get(self::FORCE_ALLOW_PROCESSING_UNPUBLISHED_ELEMENTS)) 
```

File: lib/Routing/Staticroute/Router.php [hunk 2]
```diff
@@ -245,7 +245,7 @@ protected function doMatch(string $pathinfo, Request $request = null): array
                 // to determine if a call to an action was made through a staticroute or not
                 // more on that infos see Pimcore_Controller_Action_Frontend::getRenderScript()
                 $routeParams['pimcore_request_source'] = 'staticroute';
-                $routeParams[ElementListener::FORCE_ALLOW_PROCESSING_UNPUBLISHED_ELEMENTS] = $this->config['routing']['allow_processing_unpublished_fallback_document'];
+                $routeParams[ElementListener::FORCE_ALLOW_PROCESSING_UNPUBLISHED_ELEMENTS] = true;
```
Comment (mattamon) [inline lib/Routing/Staticroute/Router.php]: Are these routeParams used anywhere? And if the value is now true anyways, couldn't we just remove this all together? Same as [here](https://github.com/pimcore/pimcore/blob/11.x/lib/Mail.php#L678) in den Mail.php right then?
[EditableUsageResolver](https://github.com/pimcore/pimcore/blob/11.x/lib/Document/Editable/EditableUsageResolver.php#L52)

**Review by mattamon — CHANGES_REQUESTED** [review]

File: lib/Routing/Staticroute/Router.php (hunk 2 above)
Comment (kingjia90) [reply lib/Routing/Staticroute/Router.php]: thank you for the feedback, it was an oversight on my part, i didn't dig to see if these const usage would be obsolete

**Review by mattamon — APPROVED** [review]

**Comment by mattamon** [comment]
@kingjia90 Can you please resolve the conflicts? :)

**Review by mattamon — APPROVED** [review]

**Pull request merged**